File: libretroshare/src/rs_android/errorconditionwrap.cpp
/*******************************************************************************
 *                                                                             *
 * libretroshare: retroshare core library                                      *
 *                                                                             *
 * Copyright (C) 2021  Gioacchino Mazzurco <gio@eigenlab.org>                  *
 * Copyright (C) 2021  Asociación Civil Altermundi <info@altermundi.net>       *
 *                                                                             *
 * This program is free software: you can redistribute it and/or modify        *
 * it under the terms of the GNU Lesser General Public License as              *
 * published by the Free Software Foundation, either version 3 of the          *
 * License, or (at your option) any later version.                             *
 *                                                                             *
 * This program is distributed in the hope that it will be useful,             *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
 * GNU Lesser General Public License for more details.                         *
 *                                                                             *
 * You should have received a copy of the GNU Lesser General Public License    *
 * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
 *                                                                             *
 *******************************************************************************/

#include "rs_android/rsjni.hpp"

namespace jni
{
Local<Object<RsJni::ErrorConditionWrap>> MakeAnything(
        ThingToMake<RsJni::ErrorConditionWrap>, JNIEnv& env,
        const std::error_condition& ec )
{
	auto& clazz = jni::Class<RsJni::ErrorConditionWrap>::Singleton(env);

	static auto method =
	        clazz.GetConstructor<jni::jint, jni::String, jni::String>(env);

	jni::jint value = ec.value();
	auto message = jni::Make<jni::String>(env, ec.message());
	auto category = jni::Make<jni::String>(env, ec.category().name());

	return clazz.New(env, method, value, message, category);
}
}

File: retroshare-service/src/android/src/org/retroshare/service/ErrorConditionWrap.java
/*
 * RetroShare
 * Copyright (C) 2021  Gioacchino Mazzurco <gio@eigenlab.org>
 * Copyright (C) 2021  Asociación Civil Altermundi <info@altermundi.net>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * SPDX-FileCopyrightText: Retroshare Team <contact@retroshare.cc>
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

package org.retroshare.service;

public class ErrorConditionWrap
{
    public ErrorConditionWrap(
        int value, String message, String categoryName )
    {
        mValue = value;
        mMessage = message;
        mCategoryName = categoryName;
    }

    public int value() { return mValue; }
    public String message() { return mMessage; }
    public String categoryName() { return mCategoryName; }

    public boolean toBool() { return mValue != 0; }

    @Override
    public String toString()
    { return String.format("%d", mValue)+" "+mMessage+" [" + mCategoryName+ "]"; }

    private int mValue = 0;
    private String mMessage;
    private String mCategoryName;
}
